lib/mongoid/association/referenced/has_and_belongs_to_many.rb

```diff
@@ -264,7 +264,11 @@ def with_ordering(criteria)
         def query_criteria(id_list)
           crit = relation_class.criteria
           crit = crit.apply_scope(scope)
-          crit = crit.all_of(primary_key => {"$in" => id_list || []})
+          crit = if id_list
+            crit.all_of(primary_key => { "$in" => id_list })
+          else
+            crit.none
```

Comment:
Could you also omit `apply_scope` call for this branch?

Reply:
sure

Reply:
done